### PR TITLE
chore: Merge master branch into feature branch

### PR DESCRIPTION
**Pull request by hengkx**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [x] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

-----
[View rendered components/form/docs/tutorial.md](https://github.com/ant-design/ant-design/blob/mergefeature1/components/form/docs/tutorial.md)
[View rendered components/popconfirm/index.en-US.md](https://github.com/ant-design/ant-design/blob/mergefeature1/components/popconfirm/index.en-US.md)
[View rendered components/popconfirm/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/mergefeature1/components/popconfirm/index.zh-CN.md)
[View rendered docs/react/contributing.zh-CN.md](https://github.com/ant-design/ant-design/blob/mergefeature1/docs/react/contributing.zh-CN.md)
[View rendered docs/react/migration-v4.en-US.md](https://github.com/ant-design/ant-design/blob/mergefeature1/docs/react/migration-v4.en-US.md)
[View rendered docs/react/use-with-create-react-app.en-US.md](https://github.com/ant-design/ant-design/blob/mergefeature1/docs/react/use-with-create-react-app.en-US.md)
[View rendered docs/react/use-with-create-react-app.zh-CN.md](https://github.com/ant-design/ant-design/blob/mergefeature1/docs/react/use-with-create-react-app.zh-CN.md)
[View rendered docs/spec/introduce.en-US.md](https://github.com/ant-design/ant-design/blob/mergefeature1/docs/spec/introduce.en-US.md)

### CODE REVIEW

**Comment by hengkx** [comment]
不行了 脑袋一片懵 要去清醒清醒了

File: components/rate/index.tsx [hunk 1]
```diff
@@ -26,50 +27,37 @@ interface RateNodeProps {
...
-    return <ConfigConsumer>{this.renderRate}</ConfigConsumer>;
-  }
-}
+  const { getPrefixCls, direction } = React.useContext(ConfigContext);
+  const { prefixCls, className, ...restProps } = props;
+  const rateProps = omit(restProps, ['tooltips']);
+  const ratePrefixCls = getPrefixCls('rate', prefixCls);
+  const rateClassNames = classNames(className, {
+    [`${ratePrefixCls}-rtl`]: direction === 'rtl',
+  });
```
Comment (afc163) [inline components/rate/index.tsx:43]: 对比一下那个 PR，这里 rateClassNames 是去掉的。

Comment (hengkx) [reply components/rate/index.tsx:43]: 你们弄把 脑袋不清醒了:sob:

Comment (hengkx) [reply components/rate/index.tsx:43]: 合一下把 我晚点在feature 上 要做些 修改

**Pull request closed without merging**